providers/google.go

```diff
@@ -229,19 +236,32 @@ func (p *GoogleProvider) setGroupRestriction(groups []string, adminEmail string,
 	}
 }
 
-func getAdminService(adminEmail string, credentialsReader io.Reader) *admin.Service {
-	data, err := io.ReadAll(credentialsReader)
-	if err != nil {
-		logger.Fatal("can't read Google credentials file:", err)
-	}
-	conf, err := google.JWTConfigFromJSON(data, admin.AdminDirectoryUserReadonlyScope, admin.AdminDirectoryGroupReadonlyScope)
-	if err != nil {
-		logger.Fatal("can't load Google credentials file:", err)
-	}
-	conf.Subject = adminEmail
-
+func getAdminService(adminEmail string, targetPrincipal string, useADC bool, credentialsReader io.Reader) *admin.Service {
```

Comment:
For better readability I would suggest to replace the provider option parameters with the options struct. this way if something new is added in the future it is directly available inside the getAdminService function.

```suggestion
func getAdminService(opts *options.GoogleOptions) *admin.Service {
```